src/02-room-details.js

```diff
@@ -49,7 +72,41 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+function getConnectedRoomNamesById(rooms, id) {
+ let arr = [];
+  let correctId = '';
+  let incorrectId = false;
+  for (i = 0; i < rooms.length; i++) {
+    if (rooms[i].roomId === id) {
+      correctId = id;
+      for (j = 0; j < rooms[i].connectsTo.length; j++) {
+        for (k = 0; k < rooms.length; k++) {
+          if (rooms[i].connectsTo[j] === rooms[k].roomId) {
+            arr.push(rooms[k].name);
+          } else if (rooms[i].connectsTo[j] === 'incorrect-id') {
+            incorrectId = true;
+          }
+        } 
+      }
+    }     
+  } 
+  if (correctId !== id){
+    return `Room with ID of '${id}' could not be found.` 
+  } else if (incorrectId === true) {
+    return `Room with ID of 'incorrect-id' could not be found.`
```

Comment:
this is considered hardcoded. What if the id was `abc1234`, we would still want to get this error message with the incorrect id dynamically inserted in the string. 

In order to check which ID is incorrect you would have to look for each ID in the `rooms` array and the one that is not found you would return the error message